From dac089515901d6bf315cd15a6e744b8d2c02c1cb Mon Sep 17 00:00:00 2001
From: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
Date: Sat, 31 Oct 2015 22:48:07 +0100
Subject: [PATCH] tfm: make a few functions static

clamav expects them to be static and it does not seem bad to do so.

Signed-off-by: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
---
 src/bin/fp_read_radix.c        | 2 +-
 src/bin/fp_read_signed_bin.c   | 2 +-
 src/bin/fp_read_unsigned_bin.c | 2 +-
 src/headers/tfm.h              | 6 +++---
 4 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/bin/fp_read_radix.c b/src/bin/fp_read_radix.c
index 0b5e826..431afa0 100644
--- a/src/bin/fp_read_radix.c
+++ b/src/bin/fp_read_radix.c
@@ -9,7 +9,7 @@
  */
 #include <tfm_private.h>
 
-int fp_read_radix(fp_int *a, char *str, int radix)
+int fp_read_radix(fp_int *a, const char *str, int radix)
 {
   int     y, neg;
   char    ch;
diff --git a/src/bin/fp_read_signed_bin.c b/src/bin/fp_read_signed_bin.c
index e2b8003..6467d19 100644
--- a/src/bin/fp_read_signed_bin.c
+++ b/src/bin/fp_read_signed_bin.c
@@ -9,7 +9,7 @@
  */
 #include <tfm_private.h>
 
-void fp_read_signed_bin(fp_int *a, unsigned char *b, int c)
+void fp_read_signed_bin(fp_int *a, const unsigned char *b, int c)
 {
   /* read magnitude */
   fp_read_unsigned_bin (a, b + 1, c - 1);
diff --git a/src/bin/fp_read_unsigned_bin.c b/src/bin/fp_read_unsigned_bin.c
index 3ee64c0..2ee89cb 100644
--- a/src/bin/fp_read_unsigned_bin.c
+++ b/src/bin/fp_read_unsigned_bin.c
@@ -9,7 +9,7 @@
  */
 #include <tfm_private.h>
 
-void fp_read_unsigned_bin(fp_int *a, unsigned char *b, int c)
+void fp_read_unsigned_bin(fp_int *a, const unsigned char *b, int c)
 {
   /* zero the int */
   fp_zero (a);
diff --git a/src/headers/tfm.h b/src/headers/tfm.h
index af87b56..f406388 100644
--- a/src/headers/tfm.h
+++ b/src/headers/tfm.h
@@ -467,14 +467,14 @@ int fp_prime_random_ex(fp_int *a, int t, int size, int flags, tfm_prime_callback
 int fp_count_bits(fp_int *a);
 
 int fp_unsigned_bin_size(fp_int *a);
-void fp_read_unsigned_bin(fp_int *a, unsigned char *b, int c);
+void fp_read_unsigned_bin(fp_int *a, const unsigned char *b, int c);
 void fp_to_unsigned_bin(fp_int *a, unsigned char *b);
 
 int fp_signed_bin_size(fp_int *a);
-void fp_read_signed_bin(fp_int *a, unsigned char *b, int c);
+void fp_read_signed_bin(fp_int *a, const unsigned char *b, int c);
 void fp_to_signed_bin(fp_int *a, unsigned char *b);
 
-int fp_read_radix(fp_int *a, char *str, int radix);
+int fp_read_radix(fp_int *a, const char *str, int radix);
 
 int fp_radix_size(fp_int *a, int radix, int *size);
 int fp_toradix(fp_int *a, char *str, int radix);
